internal/auth/oidc/auth_method.go

```diff
@@ -0,0 +1,193 @@
+package oidc
+
+import (
+	"context"
+	"crypto/ed25519"
+	"crypto/hmac"
+	"crypto/sha256"
+	"encoding/base64"
+	"fmt"
+	"net/url"
+
+	"github.com/hashicorp/boundary/internal/auth/oidc/store"
+	"github.com/hashicorp/boundary/internal/errors"
+	"github.com/hashicorp/boundary/internal/kms"
+	"github.com/hashicorp/boundary/internal/oplog"
+	wrapping "github.com/hashicorp/go-kms-wrapping"
+	"github.com/hashicorp/go-kms-wrapping/structwrapping"
+	"google.golang.org/protobuf/proto"
+)
+
+// DefaultAuthMethodTableName defines the default table name for an AuthMethod
+const DefaultAuthMethodTableName = "auth_oidc_method"
+
+// AuthMethod contains an OIDC auth method configuration. It is owned
+// by a scope.  AuthMethods can have may assigned Accounts, AudClaims,
+// CallbackUrls, Certificates, SigningAlgs.  AuthMethods also have one State at
+// any given time which determines it's behavior for many its operations.
+type AuthMethod struct {
+	*store.AuthMethod
+	tableName string
+}
+
+// NewAuthMethod creates a new in memory AuthMethod assigned to scopeId.
+// WithMaxAge, WithName and WithDescription are the only valid options. All
+// other options are ignored.
+//
+// State equals the state of the OIDC auth method.  State is not a supported
+// parameter when creating new AuthMethod's since it must be Inactive for all
+// new AuthMethods.
+//
+// DiscoveryUrl equals a URL where the OIDC provider's configuration can be
+// retrieved. The OIDC Discovery Specification requires the URL end in
+// /.well-known/openid-configuration. However, Boundary will strip the URI and
+// only store the scheme and domain.
```

Comment:
The spec only says the URL must **end** in `/.well-known/openid-configuration`. If we remove everything after the scheme and domain, won't that break all of these URLs:
- `https://bigauth.com/foo/bar/baz/.well-known/openid-configuration`
- `https://bigauth.com/bar/baz/foo/.well-known/openid-configuration`
- `https://bigauth.com/baz/foo/bar/.well-known/openid-configuration`

Reply:
ty!